contracts/vesting/VestingEscrowFactory.vy

```diff
@@ -24,80 +38,156 @@ interface VestingEscrowSimple:
     assert vesting_duration > 0  # dev: duration must be > 0
-    assert recipient not in [self, empty(address), token.address, owner] # dev: wrong recipient
+    assert recipient not in [self, empty(address), YFI.address, OWNER] # dev: wrong recipient
+    assert amount > 0
 
-    escrow: address = create_minimal_proxy_to(TARGET)
-    VestingEscrowSimple(escrow).initialize(
-        owner,
-        token,
+    idx: uint256 = self.num_vests
+    self.num_vests = idx + 1
+    self.pending_vests[idx] = Vest({
+        recipient: recipient,
+        amount: amount,
+        duration: vesting_duration,
+        start: vesting_start,
+        cliff: cliff_length
+    })
+    assert YFI.transferFrom(msg.sender, self, amount, default_return_value=True)
+
+    log VestingEscrowCreated(
+        msg.sender,
         recipient,
+        idx,
         amount,
         vesting_start,
-        vesting_start + vesting_duration,
-        cliff_length,
-        open_claim,
+        vesting_duration,
+        cliff_length
     )
 
-    # skip transferFrom and approve and send directly to escrow
-    assert token.transferFrom(msg.sender, escrow, amount, default_return_value=True)  # dev: funding failed
+    return idx
 
-    log VestingEscrowCreated(
+@external
+def deploy_vesting_contract(
+    idx: uint256,
+    token: address,
+    amount: uint256,
+    open_claim: bool = True,
+) -> (address, uint256):
+    """
+    @notice Deposit into a liquid locker and deploy a vesting contract
+    @dev Requires a vest. Requires the liquid locker being approved by the owner
+    @param idx Vest index
+    @param token Liquid locker token to deposit into
+    @param amount Amount of YFI to deposit
+    @param open_claim Whether anyone can claim 
+    @return Vest contract address, vested amount of liquid locker tokens
+    """
+    vest: Vest = self.pending_vests[idx]
+    assert msg.sender == vest.recipient
+
+    depositor: address = self.liquid_lockers[token]
+    assert depositor != empty(address)
+
+    self.pending_vests[idx].amount -= amount
+
+    assert YFI.approve(depositor, amount, default_return_value=True)
+    ll_amount: uint256 = Depositor(depositor).deposit(amount)
+    assert ll_amount > 0
+
+    escrow: address = create_minimal_proxy_to(TARGET)
+    VestingEscrowSimple(escrow).initialize(
+        OWNER,
+        token,
+        msg.sender,
+        ll_amount,
+        vest.start,
+        vest.start + vest.duration,
+        vest.cliff,
+        open_claim,
+    )
+    assert ERC20(token).transfer(escrow, ll_amount, default_return_value=True)
+    log VestingContractDeployed(
         msg.sender,
         token,
-        recipient,
+        idx,
         escrow,
         amount,
-        vesting_start,
-        vesting_duration,
-        cliff_length,
-        open_claim,
+        ll_amount,
     )
-    return escrow
\ No newline at end of file
+    return escrow, ll_amount
+
+@external
+def set_liquid_locker(liquid_locker: address, depositor: address):
```

Comment:
check depositor is not empty address?

Reply:
This is intentional, it allows factory owner to revoke permission to deposit into a liquid locker.